nipype/interfaces/base/core.py

```diff
@@ -1072,6 +1072,12 @@ def _filename_from_source(self, name, chain=None):
         if not isdefined(retval) or "%s" in retval:
             if not trait_spec.name_source:
                 return retval
+
+            # Do not generate filename when excluded by other inputs
+            if trait_spec.xor and any(isdefined(getattr(self.inputs, field))
+                                      for field in trait_spec.xor):
+                return retval
```

Comment:
@satra @oesteban Does this constitute a full fix of #2506?

Reply:
@effigies - this is a good addition, but i'm not sure this addresses #2506, which does not have any `xor`.  isn't it the case in #2506 that `out_file` should be populated if `in_files` is defined and set to some default otherwise?

i think this PR is fine, just not sure if it addresses #2506.

Reply:
Ah, right. I had it kind of backwards, but a similar fix to this should resolve #2506. I'll merge and propose a quick PR for that one.